### PR TITLE
Print warning about paths if `modal` is not found

### PR DESCRIPTION
**Pull request by erikbern**
This checks whether the `modal` binary is present on the path. If not, it outputs a warning and a link to more information (in the future we could also try to resolve the problem automatically).

This assumes the user is able to run the script using `python -m modal setup`

You can try this by deleting the `modal` binary after installation.

image

### CODE REVIEW

File: modal/cli/entry_point.py [hunk 1]
```diff
@@ -42,6 +45,25 @@ def modal(
     pass
 
 
+def setup():
+    # Sanity check the path
+    try:
+        subprocess.run(["modal", "--help"], capture_output=True)
+        # TODO(erikbern): check returncode?
```
Comment (thundergolfer) [inline modal/cli/entry_point.py:52]: Can also do `except PermissionError: ...` to handle case where file is present but not executable.

**Review by thundergolfer — APPROVED** [review]

File: modal/cli/entry_point.py (hunk 1 above)
Comment (erikbern) [reply modal/cli/entry_point.py:52]: sure good idea

**Comment by ekzhang** [comment]
Can we insert a divider before "If your browser didn't open"?

**Comment by erikbern** [comment]
> Can we insert a divider before "If the web browser didn't open"?

Yeah sounds good, it's a bit confusing.

**Comment by erikbern** [comment]
latest:

image

**Comment by erikbern** [comment]
this isn't perfect but seems like a clear pareto improvement – but let's keep iterating

File: modal/cli/entry_point.py [hunk 2]
```diff
@@ -42,6 +46,36 @@ def modal(
...
+        )
+    except PermissionError:
+        text = (
+            "[red]The `[white]modal[/white]` command is not executable!\n"
+            "You may need to give it permissions or use `[white]python -m modal[/white]` as a workaround.[/red]\n"
+        )
+    text += "See more information here:\n\n" f"[link={url}]{url}[/link]\n"
+    console = Console()
+    console.print(text)
+    console.print(Rule(style="white"))
```
Comment (ekzhang) [inline modal/cli/entry_point.py:69]: nice!

**Pull request merged**